File: python/leetcode/259.py
class Solution:
    def threeSumSmaller(self, nums, target):
        nums.sort()
        n, res = len(nums), 0
        for i in range(n - 2):
            l, r = i + 1, n - 1
            while l < r:
                if nums[i] + nums[l] + nums[r] >= target: r -= 1
                else:
                    res += r - l
                    l += 1
        return res

File: cpp/leetcode/259.cpp
class Solution {
public:
    int threeSumSmaller(vector<int>& nums, int target) {
        if(nums.size() < 3) return 0;
        sort(nums.begin(), nums.end());
        int cnts = 0;
        for(int i = 0; i < nums.size() - 2; i++){
            int l = i + 1, r = nums.size() - 1;
            while(l < r){
                if(nums[i] + nums[l] + nums[r] >= target) {
                    r--;
                } else {
                    cnts += r - l;
                    l++;
                }
            }
        }
        return cnts;
    }
};